src/Sylius/Bundle/UserBundle/Validator/Constraints/RegisteredUserValidator.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\UserBundle\Validator\Constraints;
+
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+/**
+ * @author Michał Marcinkowski <michal.marcinkowski@lakion.com>
+ */
+class RegisteredUserValidator extends ConstraintValidator
+{
+    /**
+     * @var RepositoryInterface
+     */
+    private $customerRepository;
+
+    /**
+     * @param RepositoryInterface $customerRepository
+     */
+    public function __construct(RepositoryInterface $customerRepository)
+    {
+        $this->customerRepository = $customerRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($customer, Constraint $constraint)
+    {
+        $existingCustomer = $this->customerRepository->findOneBy(['email' => $customer->getEmail()]);
```

Comment:
short array syntax is php 5.4 - we should have bumped to 5.4, but I'm not sure if it's been confirmed yet?